virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1959,143 @@ def _tarball_download(self, url, destination):
+                                       make_cmds="", stress_cmds="", stress_args="",
+                                       work_path="", uninstall_cmds="",
+                                       download_type="tarball")
+        self.vm = vm
+        self.copy_files_to = self.vm.copy_files_to
+        self.session = self.get_session()
+        self.cmd_status_output = self.session.cmd_status_output
+        self.cmd_status = self.session.cmd_status
+        self.cmd_output_safe = self.session.cmd_output_safe
+        self.sendline = self.session.sendline
 
-    @error_context.context_aware
-    def load_stress_tool(self):
+    def get_session(self):
         """
-        load stress tool on host.
+        Method to get the session of the vm instance
         """
-        # Run stress tool on host.
         try:
-            path.find_command(self.stress_type)
-        except path.CmdNotFoundError as detail:
-            raise exceptions.TestSkipError("Stress tool %s is missing: %s"
-                                           % (self.stress_type, str(detail)))
-
-        args = [self.stress_type, '--args=%s' % self.stress_args,
-                '--verbose']
-        logging.info("Start sub-process by args: %s", args)
-        self.host_stress_process = subprocess.Popen(args, universal_newlines=True,
-                                                    stdout=subprocess.PIPE,
-                                                    stderr=subprocess.PIPE)
-        running = utils_misc.wait_for(self.app_running, first=0.5, timeout=60)
-        if not running:
-            raise StressError("Stress tool %s isn't running"
-                              % self.stress_type)
-
-    @error_context.context_aware
-    def unload_stress(self):
-        """
-        stop stress tool manually
-        """
-        def _unload_stress():
-            if self.host_stress_process is not None:
-                utils_misc.kill_process_tree(self.host_stress_process.pid)
-            if not self.app_running():
-                return True
-            return False
+            session = self.vm.wait_for_login()
+            return session
+        except aexpect.ShellError as detail:
+            raise StressError("Login %s failed:\n%s" % (self.vm.name, detail))
 
-        error_context.context("stop stress app on host", logging.info)
-        utils_misc.wait_for(_unload_stress, first=2.0,
-                            text="wait stress app quit", step=1.0, timeout=60)
 
-    def app_running(self):
+class HostStress(Stress):
+
+    """
+    Run Stress tool on host, such as stress, unixbench, iozone and etc.
+    """
+
+    def __init__(self, stress_type, params, download_url="", make_cmds="",
+                 stress_cmds="", stress_args="", work_path="",
+                 uninstall_cmds="", download_type="tarball"):
+        super(HostStress, self).__init__(stress_type, params, download_url="",
```

Comment:
HostStress also have the same issue with VMStress when we call Stress.__init__()